otcextensions/osclient/dws/v1/snapshot.py

```diff
@@ -0,0 +1,306 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+#
+'''DWS Snapshot action implementations'''
+import logging
+
+from osc_lib import utils
+from osc_lib import exceptions
+from osc_lib.command import command
+from otcextensions.common import sdk_utils
+
+from otcextensions.osclient.dws.v1 import cluster as _cluster
+
+from otcextensions.i18n import _
+
+LOG = logging.getLogger(__name__)
+
+
+def _get_columns(item):
+    column_map = {
+    }
+    return sdk_utils.get_osc_show_columns_for_sdk_resource(item, column_map)
+
+
+def translate_response(func):
+    def new(self, *args, **kwargs):
+        obj = func(self, *args, **kwargs)
+        columns = (
+            'id',
+            'name',
+            'cluster_id',
+            'status',
+            'size',
+            'type',
+            'started',
+            'finished',
+        )
+
+        data = utils.get_item_properties(obj, columns)
+        return (columns, data)
+    new.__name__ = func.__name__
+    new.__doc__ = func.__doc__
+    return new
+
+
+class ListSnapshots(command.Lister):
+
+    _description = _('List DWS Backups.')
+
+    columns = (
+        'ID',
+        'Name',
+        'Cluster Id',
+    )
+
+    def get_parser(self, prog_name):
+        parser = super(ListSnapshots, self).get_parser(prog_name)
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.dws
+        data = client.snapshots()
+
+        return (self.columns, (utils.get_item_properties(s, self.columns)
+                               for s in data))
+
+
+class CreateSnapshot(command.ShowOne):
+    _description = _('Create snapshot for a specified cluster.')
+
+    def get_parser(self, prog_name):
+        parser = super(CreateSnapshot, self).get_parser(prog_name)
+
+        parser.add_argument(
+            'cluster',
+            metavar='<cluster>',
+            help=_('ID or name of the cluster for which you want to '
+                   'create a snapshot.')
+        )
+        parser.add_argument(
+            'name',
+            metavar='<name>',
+            # required=True,
```

Comment:
not required?

Reply:
Since name is a positional argument, i commented out required=True